build-system/tasks/bundle-size/index.js

```diff
@@ -38,38 +37,38 @@ const {
   VERSION: internalRuntimeVersion,
 } = require('../../compile/internal-version');
 const {cyan, green, red, yellow} = require('ansi-colors');
+const {serialReport} = require('@ampproject/filesize');
 
 const requestPost = util.promisify(require('request').post);
 
-const fileGlobs = ['dist/*.js', 'dist/v0/*-?.?.js'];
+const fileGlobs = packageJson.filesize.track;
 const normalizedRtvNumber = '1234567890123';
 
 const expectedGitHubRepoSlug = 'ampproject/amphtml';
 const bundleSizeAppBaseUrl = 'https://amp-bundle-size-bot.appspot.com/v0/';
+const replacementExpression = new RegExp(internalRuntimeVersion, 'g');
 
 /**
- * Get the brotli bundle sizes of the current build.
+ * Get the brotli bundle sizes of the current build after normalizing the RTV number.
  *
  * @return {Map<string, number>} the bundle size in KB rounded to 2 decimal
  *   points.
  */
-function getBrotliBundleSizes() {
-  // Brotli compressed size fluctuates because of changes in the RTV number, so
-  // normalize this across pull requests by replacing that RTV with a constant.
+async function getBrotliBundleSizes() {
   const bundleSizes = {};
 
   log(cyan('brotli'), 'bundle sizes are:');
-  for (const filePath of globby.sync(fileGlobs)) {
-    const normalizedFileContents = fs
-      .readFileSync(filePath, 'utf8')
-      .replace(new RegExp(internalRuntimeVersion, 'g'), normalizedRtvNumber);
-
-    const relativeFilePath = path.relative('.', filePath);
-    const bundleSize = parseFloat(
-      (brotliSize.sync(normalizedFileContents) / 1024).toFixed(2)
-    );
-    log(' ', cyan(relativeFilePath) + ':', green(`${bundleSize}KB`));
-    bundleSizes[relativeFilePath] = bundleSize;
+  const values = serialReport(process.cwd(), content =>
+    content.replace(replacementExpression, normalizedRtvNumber)
+  );
+  let next = await values.next();
+  while (!next.done) {
+    const [filePath, brotliSize] = next.value;
+    const relativePath = path.relative('.', filePath);
+    const reportedSize = parseFloat((brotliSize / 1024).toFixed(2));
+    log(' ', cyan(relativePath) + ':', green(reportedSize + 'KB'));
+    bundleSizes[relativePath] = reportedSize;
+    next = await values.next();
```

Comment:
@kristoferbaxter @danielrozenberg This loop is causing bundle sizes to be reported O(N^2) times instead of O(N).

Before: https://travis-ci.org/github/ampproject/amphtml/jobs/620244935#L581-L743
After: https://travis-ci.org/github/ampproject/amphtml/jobs/653683178#L778-L10786

Reply:
Taking a look...

Reply:
Should be fixed in https://github.com/ampproject/filesize/pull/63 // cc @kristoferbaxter

Reply:
Thanks for the quick fix! Looks good in the [latest logs](https://travis-ci.org/github/ampproject/amphtml/jobs/667468078#L561-L727).